src/deroff.rs

```diff
@@ -296,6 +328,27 @@ impl Deroffer {
             c => c.chars().all(|c| c.is_whitespace()),
         }
     }
+
+    fn not_whitespace(s: &str, idx: usize) -> bool {
```

Comment:
From what I think, we should check not whitespace for a range, not an index as the below calls it with `and`.
```rust
//          if self.s[0:2] == '\\(' and self.not_whitespace(2) and self.not_whitespace(3):
```
It could have just been
```rust
//          if self.s[0:2] == '\\(' and self.not_whitespace(2..4):
```
As well, why not just use `is_ascii_whitespace()`?

Reply:
> From what I think, we should check not whitespace for a range, not an index as the below calls it with `and`.
> 
> ```rust
> //          if self.s[0:2] == '\\(' and self.not_whitespace(2) and self.not_whitespace(3):
> ```
> 
> It could have just been
> 
> ```rust
> //          if self.s[0:2] == '\\(' and self.not_whitespace(2..4):
> ```

That's a good point. I hadn't looked closely at where `not_whitespace` was being used, but that makes sense. The one wrinkle is that the original function explicitly returns `false` if the target string is empty, which would complicate a range/iterator implementation. It would need to do a length check in addition to the character check to ensure that the actual substring length is the same as the range length. Might be something to revisit after we finish the initial port.

> As well, why not just use `is_ascii_whitespace()`?

Good question. For the initial port we're _largely_ replicating the literal behavior of the original Python source, which only checks for `' '`, `'\t'` and `'\n'`, whereas the `is_ascii_whitespace` methods in Rust return `true` for a number of other characters. I haven't looked closely enough at how the function is being used to know whether we could get away with `is_ascii_whitespace`. It's something we might be able to do after we have our port working end-to-end.

Reply:
Added a TODO to revisit ascii whitespace after MVP.